rpc/grpc/server/services/blockresultservice/service.go

```diff
@@ -0,0 +1,61 @@
+package blockresultservice
+
+import (
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/cometbft/cometbft/libs/log"
+	sm "github.com/cometbft/cometbft/state"
+	"github.com/cometbft/cometbft/store"
+
+	brs "github.com/cometbft/cometbft/proto/tendermint/services/block_results/v1"
+)
+
+type blockResultsService struct {
+	stateStore sm.Store
+	blockStore *store.BlockStore
+	logger     log.Logger
+}
+
+// New creates a new CometBFT block results service server.
+func New(bs *store.BlockStore, ss sm.Store, logger log.Logger) brs.BlockResultsServiceServer {
+	return &blockResultsService{stateStore: ss, blockStore: bs, logger: logger}
+}
+
+// GetBlockResults returns the block results of the requested height.
+// If no height is given, the block results for the latest height are returned.
+func (s *blockResultsService) GetBlockResults(_ context.Context, req *brs.GetBlockResultsRequest) (*brs.GetBlockResultsResponse, error) {
+	logger := s.logger.With("endpoint", "GetBlockResults")
+	height := req.Height
+	latestHeight := s.blockStore.Height()
+	if req.Height > latestHeight || req.Height < 0 {
+		logger.Error("Failed to validate height from BlockResults request", "height", height)
+		return nil, status.Error(codes.InvalidArgument, "Invalid request: height must be between 0 and the last effective height.")
```

Comment:
```suggestion
		return nil, status.Errorf(codes.InvalidArgument, "Height must be between 0 and the last effective height (%d)", latestHeight)
```

Might be useful to instruct the caller as to what the latest height is at the time of the request. Also, the `Invalid request` part of the message isn't really necessary given the error code we're sending back.